onset_maproom/cess.py

```diff
@@ -0,0 +1,129 @@
+#temporary file for cessation date functions; 
+#these will be moved to the main calc.py file when finished
+
+import calc
+import numpy as np
+import pandas as pd
+import xarray as xr
+
+#params for cess as found in maproom
+search_start_day = 1
+search_start_month = 9
+search_days = 90
+dry_thresh = 5
+min_dry_days = 3
+time_coord = "T"
+et = 5
+taw = 60
+sminit = 0 
+
+rr_mrg = calc.read_zarr_data("/data/aaron/ethiopia-rain-rechunked")
+
+#subset of data for testing
+rr_mrgSub = rr_mrg.sel(T=slice('1981-01-01','1983-12-31'))
+daily_rain = rr_mrgSub.isel(X=200,Y=160)
+
+
+#onset date calc used in seasonal cess date func
+#first date after START_DAY(1), START_MONTH(9) in SEARCH_DAYS(90) 
+#days when the soil water balance falls below DRY_THRESH(5)mm 
+#for a period of DRY_SPELL_LENGTH(3) days
+def cess_date(daily_rain, dry_thresh, min_dry_days, et, taw, sminit, time_coord="T"):
+    water_balance = calc.water_balance(daily_rain, et,taw,sminit,"T")
+    dry_day = water_balance < dry_thresh
+    first_dry_day = dry_day * 1
+    first_dry_day_roll = first_dry_day.rolling(**{time_coord: min_dry_days}).sum() >= min_dry_days
+    cess_mask = first_dry_day_roll * 1
+    cess_mask = cess_mask.where((cess_mask == 1))
+    cess_delta = cess_mask.idxmax(dim=time_coord)
+    cess_delta = cess_mask.idxmax(dim=time_coord)
+    cess_delta = (
+        cess_delta
+    # offset relative position of first wet day
+    # note it doesn't matter to apply max or min
+    # per construction all values are nan but 1
+        - (
+            min_dry_days
+            - 1
+            - first_dry_day.where(first_dry_day[time_coord] == cess_delta).max(
```

Comment:
I just pushed changes however for this, I was having a difficult time subtracting (min_dry_days - 1) from cess_delta; it kept subtracting nanoseconds. My attempts to fix it so far have not been successful except to leave in this line in lieu of a constant 1 

`dry_spell.where(dry_spell[time_coord] == cess_delta).max(
                dim=time_coord)`

Since that will always give a value of 1 anyways..

Not necessarily the smartest fix but for now it worked so I went with it..